CHANGELOG.md

```diff
@@ -10,12 +10,318 @@ New features:
+
+Previously one could do:
+
+```purescript
+foreign import kind Boolean
+foreign import data True :: Boolean
+foreign import data False :: Boolean
+```
+
+Where the kind `Boolean` and type `Boolean` were two different things. This is no longer the case. The `Prim` kind `Boolean` is now removed, and you can just use `Prim` type `Boolean` in the same way. This is a breaking change.
+
+The compiler still supports the old `foreign import kind` syntax but it warns that it's deprecated.
+
+```purescript
+foreign import kind Foo
+```
+
+> Foreign kind imports are deprecated and will be removed in a future release. Use empty 'data' instead.
+
+It is treated internally as:
+
+```purescript
+data Foo
+```
+
+Likewise, `kind` imports and exports are deprecated and treated the same as a type import or export.
+
+> Kind imports are deprecated and will be removed in a future release. Omit the 'kind' keyword instead.
+
+The special unary `#` syntax for row kinds is still supported, but deprecated and will warn. There is now `Prim.Row :: Type -> Type` which can be used like a normal type constructor.
+
+> Unary '#' syntax for row kinds is deprecated and will be removed in a future release. Use the 'Row' kind instead.
+
+All of these deprecations have suggested fixes in the JSON output, so tools like [`purescript-suggest`](https://github.com/nwolverson/purescript-suggest) (or your IDE plugin) can automatically apply them.
+
+#### Kind Signatures
+
+With PolyKinds, all type-level declarations are generalized.
+
+```purescript
+data Proxy a = Proxy
+```
+
+Previously, this had the `Type`-defaulted kind `Type -> Type`. Now this will be generalized to `forall k. k -> Type`. Such signature can be written with a kind signature declarations, similar to [standalone kind signatures](https://ryanglscott.github.io/2020/01/05/five-benefits-to-using-standalonekindsignatures/) in GHC.
+
+```purescript
+data Proxy :: forall k. k -> Type
+data Proxy a = Proxy
+```
+
+In GHC, all signatures use the `type` prefix, but we reuse the same keyword as the subsequent declaration because we already have `foreign import data` (rather than `foreign import type`) and because it makes things align nicer. Signatures have the same rule as value-level signatures, so they must always be followed by the "real" declaration.
+
+It's better to be explicit about polymorphism by writing signatures. Since we don't really quantify over free type variables, it's also necessary in the case that two poly-kinded arguments must have the same kind. The compiler will warn about missing kind signatures when polymorphic kinds are inferred.
+
+Classes can have signatures too, but they must end with the new `Constraint` kind instead of `Type`. For example, `Prim.Row.Cons` now has the signature:
+
+```purescript
+class Cons :: forall k. Symbol -> k -> Row k -> Row k -> Constraint
```

Comment:
Previously, `Cons` had [this type](https://pursuit.purescript.org/builtins/docs/Prim.Row#t:Cons):

```purs
class Cons (label :: Symbol) (a :: k) (tail :: Row k) (row :: Row k) | label a tail -> row, label row -> a tail
```

Now, it has this type and also a kind signature, correct?


```purs
class Cons :: forall k. Symbol -> k -> Row k -> Row k -> Constraint
class Cons (label :: Symbol) (a :: k) (tail :: Row k) (row :: Row k) | label a tail -> row, label row -> a tail
```

Perhaps it can also be written this way, now that the kind signature is present:

```purs
class Cons :: forall k. Symbol -> k -> Row k -> Row k -> Constraint
class Cons label a tail row | label a tail -> row, label row -> a tail
```

It might be useful to include this as a more complete example.

Reply:
The documentation on Pursuit is confusing. The following class declaration fails to compile because the kind variable isn’t bound:

```purs
class Cons (label :: Symbol) (a :: k) (tail :: Row k) (row :: Row k) | label a tail -> row, label row -> a tail
```

```
Error found:
in module Example
at Example.purs:3:36 - 3:37 (line 3, column 36 - line 3, column 37)

  Type variable k is undefined.

while inferring the kind of k
while checking that type k
  has kind Type
in type synonym Cons$Dict

See https://github.com/purescript/documentation/blob/master/errors/UndefinedTypeVariable.md for more information,
or to contribute content related to this error.
```

It isn’t possible to bind the kind variable with a kind signature, meaning that the following still fails to compile:

```purs
class Cons :: forall k. Symbol -> k -> Row k -> Row k -> Constraint
class Cons (label :: Symbol) (a :: k) (tail :: Row k) (row :: Row k) | label a tail -> row, label row -> a tail
```

The kind annotations in the class head must be scrapped, as in your last snippet. The kind signature is not present in the generated documentation though:

<img width="701" alt="Capture d’écran 2021-01-23 à 15 44 00" src="https://user-images.githubusercontent.com/2788549/105588814-e9422780-5d91-11eb-93e3-36e862e40efe.png">

Should we display kind signatures in Pursuit when they contain polymorphic kinds? Something like 

<img width="684" alt="Capture d’écran 2021-01-23 à 15 51 04" src="https://user-images.githubusercontent.com/2788549/105592407-e5fb6b80-5d92-11eb-8992-6bd8b6b19af6.png">

Reply:
Ideally we could update Pursuit to display kind signatures containing polymorphic kinds, but I'm not sure it's an easy task 😅

Reply:
I think we should make updating the HTML docs generation to handle polykinds a priority post-0.14.0. I've usually regretted not being more of a stickler for this in the past when we have added new language features: any information which forms part of a module's API should be present in HTML docs, and I think in an ideal world we would actually not permit merging PRs which add language features until those features are represented properly in HTML docs.

Reply:
I just found out you already opened https://github.com/purescript/purescript/issues/3973 some time ago. Let's tackle this once 0.14 is out then.